lib/list_validator.rb

```diff
@@ -0,0 +1,11 @@
+class ListValidator
+  def initialize(list)
+    @list = list
+  end
+
+  def all_valid?(elements)
+    elements.present? &&
```

Comment:
We could potentially switch:
`elements.present? && elements.is_a?(Array) && ...`
with:
`elements.is_a?(Array) && elements.any? && ...`

...if you wanted to be a bit clearer about why we're checking `.present?`, but I only mention it as something we'd discussed.